tests/upload/test_views.py

```diff
@@ -16,21 +17,48 @@ def antivirus(mocker):
     return mocker.patch('app.upload.views.antivirus_client')
 
 
-def test_document_upload_returns_link_to_frontend(client, store, antivirus):
+def _document_upload(client, url, file_content, content_type):
+    if content_type == 'multipart/form-data':
+        response = client.post(
+            url,
+            content_type=content_type,
+            data={
+                'document': (io.BytesIO(file_content), 'file.pdf')
+            }
+        )
+    if content_type == 'application/json':
+        response = client.post(
+            url,
+            json={
+                'document': base64.b64encode(file_content).decode('utf-8'),
+            }
+        )
+    return response
+
+
+@pytest.mark.parametrize(
+    'content_type',
+    (
+        'multipart/form-data',
+        'application/json',
+    )
+)
+def test_document_upload_returns_link_to_frontend(client, store, antivirus, content_type):
     store.put.return_value = {
         'id': 'ffffffff-ffff-ffff-ffff-ffffffffffff',
         'encryption_key': bytes(32),
     }
 
     antivirus.scan.return_value = True
 
-    response = client.post(
-        '/services/00000000-0000-0000-0000-000000000000/documents',
-        content_type='multipart/form-data',
-        data={
-            'document': (io.BytesIO(b'%PDF-1.4 file contents'), 'file.pdf')
-        }
-    )
+    url = '/services/00000000-0000-0000-0000-000000000000/documents'
+    file_content = b'%PDF-1.4 file contents'
+    response = _document_upload(client, url, file_content, content_type)
+
+    # Check that the contents of the file saved is as expected
+    put_args, put_kwargs = store.put.call_args_list[0]
```

Comment:
It's a little unfortunate that we have to interrogate mock calls here, but I felt that it was important to have certainty that files end up saved as expected for calls to the API through either content type.